File: src/AI_Driven_PJ.Domain/Entities/Bank.cs
using System;
using System.Collections.Generic;

namespace AI_Driven_PJ.Domain.Entities;

public partial class Bank
{
    public int BankId { get; set; }

    public string BankName { get; set; } = null!;

    public string BankCode { get; set; } = null!;

    public string Status { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public string CreatedBy { get; set; } = null!;

    public DateTime? UpdatedAt { get; set; }

    public string? UpdatedBy { get; set; }
}

File: src/AI_Driven_PJ.Client/src/app/features/banks/pages/models/bank.ts
export interface BankModel {
  bankId: number;
  bankName: string;
  bankCode: string;
  status: BankStatus;
  createdAt: string;
  createdBy: string;
  updatedAt: string | null;
  updatedBy: string | null;
}

export type BankStatus = 'ACTIVE' | 'INACTIVE';

export interface BankRequestModel {
  bankName: string;
  bankCode: string;
  status: BankStatus;
}

export interface BankQueryParams {
  page: number;
  take: number;
  search?: string;
  sortField?: string;
  sortOrder?: number;
}

export interface PagedResultModel<T> {
  items: T[];
  page: number;
  take: number;
  totalCount: number;
  totalPages: number;
}
